pkg/agent/run.go

```diff
@@ -26,70 +24,11 @@ import (
 	"github.com/jetstack/preflight/api"
 	"github.com/jetstack/preflight/pkg/client"
 	"github.com/jetstack/preflight/pkg/datagatherer"
-	"github.com/jetstack/preflight/pkg/kubeconfig"
 	"github.com/jetstack/preflight/pkg/logs"
 	"github.com/jetstack/preflight/pkg/version"
 )
 
-// ConfigFilePath is where the agent will try to load the configuration from
-var ConfigFilePath string
-
-// Period is the time waited between scans
-var Period time.Duration
-
-// OneShot flag causes agent to run once
-var OneShot bool
-
-// VenafiCloudMode flag determines which format to load for config and credential type
-var VenafiCloudMode bool
-
-// ClientID is the clientID in case of Venafi Cloud mode
-var ClientID string
-
-// PrivateKeyPath is the path for the service account private key in case of Venafi Cloud mode
-var PrivateKeyPath string
-
-// CredentialsPath is where the agent will try to loads the credentials. (Experimental)
-var CredentialsPath string
-
-// OutputPath is where the agent will write data to locally if specified
-var OutputPath string
-
-// InputPath is where the agent will read data from instead of gathering from clusters if specified
-var InputPath string
-
-// BackoffMaxTime is the maximum time for which data gatherers will be retried
-var BackoffMaxTime time.Duration
-
-// StrictMode flag causes the agent to fail at the first attempt
-var StrictMode bool
-
-// APIToken is an authentication token used for the backend API as an alternative to oauth flows.
-var APIToken string
-
-// VenConnName is the name of the VenafiConnection resource to use. Using this
-// flag will enable Venafi Connection mode.
-var VenConnName string
-
-// VenConnNS is the namespace of the VenafiConnection resource to use. It is
-// only useful when the VenafiConnection isn't in the same namespace as the
-// agent.
-//
-// May be left empty to use the same namespace as the agent.
-var VenConnNS string
-
-// InstallNS is the namespace in which the agent is running in. Only needed when
-// running the agent outside of Kubernetes.
-//
-// May be left empty when running in Kubernetes. In this case, the namespace is
-// read from the file /var/run/secrets/kubernetes.io/serviceaccount/namespace.
-var InstallNS string
-
-// Profiling flag enabled pprof endpoints to run on the agent
-var Profiling bool
-
-// Prometheus flag enabled Prometheus metrics endpoint to run on the agent
-var Prometheus bool
+var Flags AgentCmdFlags
```

Comment:
**Self-review:** This variable is public so that the `cmd/agent.go` file can import the global var `Flags` to register the command line flags.